File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/tuple/DtoTuple.java
package org.babyfish.jimmer.sql.tuple;

import org.babyfish.jimmer.sql.TypedTuple;
import org.babyfish.jimmer.sql.model.dto.BookViewForTupleTest;

@TypedTuple
public class DtoTuple {

    private BookViewForTupleTest book;

    private long authorCount;

    public BookViewForTupleTest getBook() {
        return book;
    }

    public void setBook(BookViewForTupleTest book) {
        this.book = book;
    }

    public long getAuthorCount() {
        return authorCount;
    }

    public void setAuthorCount(long authorCount) {
        this.authorCount = authorCount;
    }
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/query/tuple/DtoTuple.kt
package org.babyfish.jimmer.sql.kt.query.tuple

import org.babyfish.jimmer.sql.TypedTuple
import org.babyfish.jimmer.sql.kt.model.classic.book.dto.BookViewForTupleTest

@TypedTuple
data class DtoTuple(
    val book: BookViewForTupleTest,
    val authorCount: Long
)